Racerlib

Victor-Savu opened the pull request:
First step towards phildawes/racer#94. The restructuring was all done at high level, so it only involved moving files around and fixing the imports.

phildawes commented:
Hi @Victor-Savu. Cool! I think you forgot to git-add the bin and test directories though

Victor-Savu commented:
Thanks! I didn't notice that bin was in the .gitignore. I suggest removing it from there since it is the customary place to put the sources for executable targets.

The tests I think I have added. I just moved the system tests to their own directory. I was following the guidelines from the book (or at least as much as I could remember without actually reading it again). I don't have a strong opinion for where the tests should be, but it makes the separation between crate and module-level testing more clear. Going forwards, it might also help maintain the growing test suite.

Victor-Savu commented:
Whoops! I managed to break some tests when I added the extra level of indentation for the test module. Should be fixed now.

phildawes commented:
Looks good to me, merged. Thanks v much @Victor-Savu
